deploy/kubernetes/elastic-agent-kustomize/default/README.md

```diff
@@ -0,0 +1,69 @@
+
+The list below includes the official [kustomize](https://github.com/kubernetes-sigs/kustomize) templates to run them in Kubernetes:
+
+Agent Scenario | Description
+---- | ----
+[Elastic Agent managed - Default ](./elastic-agent-managed/) | Default Elastic Agent managed by Fleet setup. Kube-state-metrics (KSM) is installed automatically.
+[Elastic Agent standalone Default ](./elastic-agent-standalone/) | Default Standalone Elastic Agent setup. Kube-state-metrics (KSM) is installed automatically.
+
+## Using above templates
+
+Users can clone this repository to use the provided kustomize templates.
+
+For *Managed Elastic Agent*, please update the following variables inside main kustomization.yaml:
+
+- %FLEET_URL%: Fleet Server URL to enroll the Elastic Agent into. FLEET_URL can be found in Kibana, go to Management > Fleet > Settings
+- %ENROLLMENT_TOKEN%: Elasticsearch API key used to [enroll Elastic Agents](https://www.elastic.co/guide/en/fleet/current/fleet-enrollment-tokens.html#fleet-enrollment-tokens) in Fleet. *This should be encoded as base64 value because it will be stored as Kubernetes secret*
+
+Eg.
+
+```yaml
+secretGenerator:
+  - name: elastic-agent-creds
+    literals:
+      - enrollment_token=%ENROLLMENT_TOKEN%
+```
+
+For *Standalone Elastic Agent*, please update the following secrets inside main [kustomization.yaml](./elastic-agent-managed/kustomization.yaml):
+
+- %ES_HOST%: The Elasticsearch host to communicate with
+- %API_KEY: The API Key with access privileges to connect to Elasticsearch. See [create-api-key-standalone-agent](https://www.elastic.co/guide/en/fleet/current/grant-access-to-elasticsearch.html#create-api-key-standalone-agent). *This should be encoded as base64 value because it will be stored as Kubernetes secret*
+- %CA_TRUSTED%: The ssl.ca_trusted_fingerprint in order the elastic agent to be able to trust the certificate authority of the Elasticsearch output.
+- %DATASET_ID%: A string that will be added as a new field and will denote a specific installation. *By default, this will be added to state_pod dataset.*
+
+## Remote usage of kustomize templates
+
+Users can use following commands:
+
+Managed Elastic Agent:
+
+```bash
+❯ kubectl https://github.com/elastic/elastic-agent/deploy/kubernetes/elastic-agent-kustomize/default/elastic-agent-maanged\?ref\=main | sed -e "s/JUVOUk9MTE1FTlRfVE9LRU4l/base64_ENCODED_ENROLLMENT_TOKEN/g" -e "s/%FLEET_URL%/https:\/\/localhost:9200/g" | kubectl apply -f-
+
+```
+
+Standalone Elastic Agent:
+
+```bash
+kubectl kustomize https://github.com/elastic/elastic-agent/deploy/kubernetes/elastic-agent-kustomize/default/elastic-agent-standalone\?ref\=main | sed -e "s/JUFQSV9LRVkl/<base64_encoded_APIKEY>/g" -e "s/%ES_HOST%/https:\/\/localhost:9200/g" -e "s/%CA_TRUSTED%/ca_trusted_fingerprint/g" -e "s/%DATASET_ID%/12345/g" | kubectl apply -f-
+```
+
+Examples of Base64 encoded values:
+
+```bash
+❯ echo %API_KEY | base64
+JUFQSV9LRVkK
+
+echo %ENROLLMENT_TOKEN% | base64
```

Comment:
here should be used `echo -n` instead, otherwise it might not work:
```
% echo -n "Zlh5ekw1QUJubU8xdUFJajNpS2M6QWdaZFIwX0xUNmFhQ3BGbUVxTVZwUQ==" | base64
WmxoNWVrdzFRVUp1YlU4eGRVRkphak5wUzJNNlFXZGFaRkl3WDB4VU5tRmhRM0JHYlVWeFRWWndVUT09

% echo "Zlh5ekw1QUJubU8xdUFJajNpS2M6QWdaZFIwX0xUNmFhQ3BGbUVxTVZwUQ==" | base64
WmxoNWVrdzFRVUp1YlU4eGRVRkphak5wUzJNNlFXZGFaRkl3WDB4VU5tRmhRM0JHYlVWeFRWWndVUT09Cg==
```

when the result of `echo <token> | base64` is used, agent fails with error:
```
Error: fail to enroll: fail to execute request to fleet-server: net/http: invalid header field value for "Authorization"
```

ideally would be to avoid this manual transformation, wondering if kustomize does not provide this option somehow

Reply:
@tetianakravchenko  I update the README file for this. Indeed echo adds an extra line by default. 

During the onboarding flow the base64 values are provided from the kibana ui page